### PR TITLE
swagger-ui module

### PR DESCRIPTION
**Pull request by RafalSumislawski**
Closes: #366

### CODE REVIEW

File: swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+object SwaggerUiRoutes {
+
+  def apply[F[+ _] : Sync : ContextShift](swaggerUrl: String, blocker: Blocker): SwaggerUiRoutes[F] = {
+    val swaggerUiResourcesPath = s"/META-INF/resources/webjars/swagger-ui/${BuildInfo.swaggerUiVersion}/"
+    val indexHtml = defaultIndexHtml(swaggerUrl)
+    new SwaggerUiRoutes[F](swaggerUiResourcesPath, indexHtml, blocker)
+  }
+
+  case class Html(html: String)
```
Comment (zarthross) [inline swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala]: I generally like like the way a good type alias can impact the type signature and make it more readable, but is this buying us anything here?  Generally we will only define the swagger-ui once, we aren't checking that the indexHtml is valid like what might be done with an opaque alias...

**Comment by RafalSumislawski** [comment]
The pulling out of SwaggerMetadata didn't affect directly the `org.http4s.rho.swagger.models.Swagger`. It still contains all the `info`, `host`, for which there's no good Monoid definition. 

The way `SwaggerModelsBuilder` is implemented shows, that there should be some Monoid hiding in the `Swagger`.  I think the pair of paths and definitons, let's call it `PathsAndDefinitions` could form a well defined monoid. `SwaggerModelsBuilder` could operate on `PathsAndDefinitions`, and later we could merge the `PathsAndDefinitions` with `SwaggerMetadata ` to get a `Swagger`.

The old URL to swagger-ui should work and redirect to /swagger-ui/index.html. It doesn't, and it has something to do with prepending `swaggerUiTypedPath` in createRhoMiddleware. I'll debug it.

File: swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala (hunk 1 above)
Comment (RafalSumislawski) [reply swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala]: The sole purpose of the `Html` is to create a new type for the implicit search of `EntityEncoder`. The indexHtml is a `String`, but it needs a custom encoder (with html content type). 

Of course there are different ways of achieving that goal, for example:
```
  private val htmlEncoder: EntityEncoder[F, String] =
    EntityEncoder.stringEncoder[F]
      .withContentType(`Content-Type`(MediaType.text.html).withCharset(org.http4s.Charset.`UTF-8`))

  GET / "index.html" |>> { () =>
    Ok(indexHtml)(implicitly, htmlEncoder)
  }
```
would be an alternative way of doing it.

Comment (zarthross) [reply swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala]: That makes sense.  I think I would bias this explicit approach as to not burden users with the wrapping... especially for something that is used only once like this.

Comment (RafalSumislawski) [reply swagger-ui/src/main/scala/com/http4s/rho/swagger/ui/SwaggerUiRoutes.scala]: Ok for me.

**Comment by RafalSumislawski** [comment]
Hello @zarthross 
Is there something more I should do before we can merge this one?

**Pull request merged**